group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,574 @@
+            Collections.emptyMap())
+        );
+
+        members.put(consumerB, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic3Uuid),
+            Collections.emptyMap())
+        );
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<Uuid, Set<Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        mkAssignment(expectedAssignment, topic1Uuid, Arrays.asList(0, 1));
+        mkAssignment(expectedAssignment, topic1Uuid, Collections.singleton(2));
+        // Topic 3 Partitions Assignment
+        mkAssignment(expectedAssignment, topic3Uuid, Collections.singleton(0));
+        mkAssignment(expectedAssignment, topic3Uuid, Collections.singleton(1));
+
+        assertAssignment(expectedAssignment, computedAssignment);
+    }
+
+    @Test
+    public void testFirstAssignmentThreeConsumersThreeTopicsDifferentSubscriptions() {
+        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        topics.put(topic1Uuid, new AssignmentTopicMetadata(3));
+        topics.put(topic2Uuid, new AssignmentTopicMetadata(3));
+        topics.put(topic3Uuid, new AssignmentTopicMetadata(2));
+
+        Map<String, AssignmentMemberSpec> members = new HashMap<>();
+        // Initial Subscriptions: A -> T1, T2 | B -> T3 | C -> T2, T3
+
+        members.put(consumerA, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            Collections.emptyMap())
+        );
+
+        members.put(consumerB, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Collections.singletonList(topic3Uuid),
+            Collections.emptyMap())
+        );
+
+        members.put(consumerC, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic2Uuid, topic3Uuid),
+            Collections.emptyMap())
+        );
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<Uuid, Set<Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        mkAssignment(expectedAssignment, topic1Uuid, Arrays.asList(0, 1, 2));
+        // Topic 2 Partitions Assignment
+        mkAssignment(expectedAssignment, topic2Uuid, Arrays.asList(0, 1));
+        mkAssignment(expectedAssignment, topic2Uuid, Collections.singleton(2));
+        // Topic 3 Partitions Assignment
+        mkAssignment(expectedAssignment, topic3Uuid, Collections.singleton(0));
+        mkAssignment(expectedAssignment, topic3Uuid, Collections.singleton(1));
+
+        assertAssignment(expectedAssignment, computedAssignment);
```

Comment:
This is a perfect example to illustrate my previous comment. In this case, `consumerA` cannot get `topic3Uuid` but we don't really verify this.

Reply:
this was my concern too which is why I had asked for advice and this was the best idea we had all come up with, but I like the treeMap idea I'm gonna go ahead and do that

Reply:
I verified with print statements so there's no issue with the code however jic that was also a concern

Reply:
I understand that the code is doing the right thing. What I meant is that the assertions would not catch all issues.

Reply:
changing it